File: application/smartgarden_flutter_socket/lib/src/models/command_log.dart
class CommandLog {
  final int nodeId;
  final String command;
  final String status;
  final String topic;
  final DateTime createdAt;

  CommandLog({
    required this.nodeId,
    required this.command,
    required this.status,
    required this.topic,
    required this.createdAt,
  });

  factory CommandLog.fromJson(Map<String, dynamic> j) => CommandLog(
        nodeId: (j['node_id'] as num).toInt(),
        command: j['command'] as String,
        status: j['status'] as String,
        topic: j['topic'] as String,
        createdAt: DateTime.parse(j['createdAt'] as String),
      );
}

File: server/src/v1/models/CommandLog.js
import mongoose from "mongoose";

const CommandLogSchema = new mongoose.Schema(
  {
    node_id: { type: Number, required: true, index: true },
    command: { type: String, enum: ["ON", "OFF"], required: true },
    topic: { type: String, required: true },
    by: { type: String, default: "api" }, // who issued the command
    status: { type: String, enum: ["sent", "failed"], default: "sent" },
    error: { type: String, default: "" }
  },
  { timestamps: { createdAt: "createdAt", updatedAt: false } }
);

export default mongoose.model("CommandLog", CommandLogSchema);
